sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/EventingTestKit.java

```diff
@@ -40,17 +40,73 @@ static EventingTestKit start(ActorSystem system, String host, int port, MessageC
     return EventingTestKitImpl.start(system, host, port, codec);
   }
 
-  Topic getTopic(String topic);
-
   /**
-   * Testkit utility to mock broker's topic. Useful when doing integration tests for services that do eventing (in or out) to a broker's topic.
-   *
-   * <p><b>Note: </b> messages written to the topic with this utility are not readable with the expect* methods,
-   * unless they have been properly forwarded through an eventing.out flow to the same topic.
+   * Use <code>getTopicDestination</code> or <code>getTopicSubscription</code> instead.
    */
-  @ApiMayChange
-  interface Topic {
+  @Deprecated
+  Topic getTopic(String topic);
+
+  MockedDestination getTopicDestination(String topic);
+
+  MockedSubscription getTopicSubscription(String topic);
+
+  MockedSubscription getValueEntitySubscription(String typeId);
 
+  MockedSubscription getEventSourcedSubscription(String typeId);
+
+  MockedSubscription getStreamSubscription(String service, String streamId);
+
+  interface MockedSubscription {
```

Comment:
Topic methods for publishing and asserting are duplicated (Topic will be removed later) and divided into MockedSubsription and MockedDestination.